c/tests/test_minimal_cpp.cpp

```diff
@@ -78,12 +80,163 @@ test_table_basics()
     tsk_table_collection_free(&tables);
 }
 
+/* A definition of sort_edges that uses C++ std::sort and inlining of the
+ * comparison function to achieve significantly better performance than
+ * the builtin method in tskit.
+ */
+int
+cpp_sort_edges(tsk_table_sorter_t *sorter, tsk_size_t start)
+{
+    struct _edge {
+        double left, right;
+        tsk_id_t parent, child;
+
+        _edge(double l, double r, tsk_id_t p, tsk_id_t c)
+            : left{ l }, right{ r }, parent{ p }, child{ c }
+        {
+        }
+    };
+    tsk_edge_table_t *edges = &sorter->tables->edges;
+    const double *node_time = sorter->tables->nodes.time;
+    std::vector<_edge> sorted_edges;
+    size_t num_edges = edges->num_rows;
+    size_t j;
+
+    /* This is the comparison function.  We cannot define an
+     * operator < for _edge because we need to bind the node times
+     * so we have to use a functional method. This is a copy of the cmp
+     * from fwdpp.  Only difference is the final time comparison
+     * (fwdpp table times to forwards). */
```

Comment:
My bad: ``fwdpp table times go forwards``.

Reply:
Ah, now I understand, my brain didn't parse the typo.  :->

Reply:
Thanks Kevin, fixed.